cmd/config.go

```diff
@@ -39,7 +39,7 @@ func (pc *PasswordConfig) Pass() (string, error) {
 // be embedded in other config structs.
 type ServiceConfig struct {
 	// DebugAddr is the address to run the /debug handlers on.
-	DebugAddr string `validate:"hostname_port"`
```

Comment:
From a CLI UX perspective, I think either of these options would be helpful for the user
*)  A startup log should be emitted if both a flag and the related configuration file are setting a host:port and continue running the process
*) Emit the log line, but prevent the boulder component from running because it might surprise the user that metrics or grpc were available on an unanticipated port and they should clean up their config/flags.

Relatedly, it would be helpful if `//cmd/shell.go` emitted a log for the listening port(s).